### PR TITLE
Get a barebones version of workable code with a single example that c…

### PR DESCRIPTION
**Pull request by wolfwoolford**
…ompiles, to kick things off

### CODE REVIEW

**Comment by wolfwoolford** [comment]
@ncoghlan I'm not sure of the etiquette here.. I assigned to you is that how you want to proceed or please advise otherwise.

**Comment by ncoghlan** [comment]
@wolfwoolford I'm happy for you to merge your own changes directly (that's why I sent you a contributor invite).

At the moment, transferring code from here to the internal build tree is manual, and I suspect it will likely stay that way in the future (i.e. the part we'll actually use from the GitHub repo is the header file generator, not the example header files).

File: src/cereal/schema_enum.hpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        schema_enum_value  ssv;
+        ssv.v_ = SchemaEnum::verify(value);
+        return ssv;
+    }
+
+    schema_enum_value& operator=(const std::string& v)
+    {
+        *this = from_string(v);
+        return *this;
+    }
```
Comment (ncoghlan) [inline src/cereal/schema_enum.hpp]: That reminds me, there are a couple of convenience enhancements I want to copy out of the prototype (most notably, a simple "operator==" implementation for direct comparisons with const std::string references)

Comment (wolfwoolford) [reply src/cereal/schema_enum.hpp]: Sounds good to me!

**Comment by ncoghlan** [comment]
Any objections we might end up having would be in the form of a PR that we'd ask you to take a look at :)

For the initial prototype, I just stuck in a manual string -> datetime reformatting call, but I agree it would be much nicer to deal with it as part of the code generator.

**Comment by wolfwoolford** [comment]
:)

fyi I've built some code to handle rfc3339 formatted time strings and the latest push uses it. It might be useful to you. I'd like to get some unit tests in, to verify it's good, but my interactive testing of the code told me it looked pretty good.

There a few little things I want to make more consistent, like commenting. I've got a TODO list in there that represents my general thinking fyi. Maybe I should turn it into issues?

I'm assuming my pushing is ok. Let me know at anytime if you think I push something that should go in branch first. :)

**Pull request merged**